src/services/indexer.ts

```diff
@@ -19,473 +19,467 @@
-
-    // TODO: Configure ES request concurrency
-    await PromisePool.withConcurrency(20)
-      .for(sortedFiles)
-      .process(async programData => {
-        const body = programData.files.map(file => ({ delete: { _id: file.objectId } }));
-        const index = await getNextIndex(programData.program, {
-          isPublic: true,
-          clone: true,
-        });
-
-        try {
-          await client.bulk({
-            index,
-            body,
-          });
-        } catch (e) {
-          logger.error(`Failed bulk delete request: ${JSON.stringify(e)}`, e);
-          throw e;
-        }
-      });
-  }
-
-  async function removeFilesFromRestricted(files: File[]): Promise<void> {
-    const sortedFiles = sortFilesIntoPrograms(files);
-
-    // TODO: Configure ES request concurrency
-    await PromisePool.withConcurrency(20)
-      .for(sortedFiles)
-      .process(async programData => {
-        const body = programData.files.map(file => ({ delete: { _id: file.objectId } }));
-        const index = await getNextIndex(programData.program, {
-          isPublic: false,
-          clone: true,
-        });
-
-        try {
-          await client.bulk({
-            index,
-            body,
-          });
-        } catch (e) {
-          logger.error(`Failed bulk delete request: ${JSON.stringify(e)}`, e);
-          throw e;
-        }
-      });
-  }
-
-  async function deleteIndices(indices: string[]): Promise<void> {
-    if (!indices || !indices.length) {
-      return;
-    }
-    await client.indices.delete({ index: indices });
-
-    // remove the nextIndices and currentIndices references to these indexNames
-    indices.forEach(indexName => {
-      for (const programId in currentIndices.public) {
-        if (currentIndices.public[programId].indexName === indexName) {
-          delete currentIndices.public[programId];
-        }
-      }
-      for (const programId in currentIndices.restricted) {
-        if (currentIndices.restricted[programId].indexName === indexName) {
-          delete currentIndices.restricted[programId];
-        }
-      }
-      for (const programId in nextIndices.public) {
-        if (nextIndices.public[programId].indexName === indexName) {
-          delete nextIndices.public[programId];
-        }
-      }
-      for (const programId in nextIndices.restricted) {
-        if (nextIndices.restricted[programId].indexName === indexName) {
-          delete nextIndices.restricted[programId];
-        }
-      }
-    });
-  }
-
-  return {
-    // By FileDocument
-    indexRestrictedFileDocs,
-    removeRestrictedFileDocs,
-
-    updateRestrictedFile,
-
-    // Public Index Management
-    createEmptyPublicIndices,
-    createEmptyRestrictedIndices,
-    indexPublicFileDocs,
-    deleteIndices,
-    removeFilesFromPublic,
-    removeFilesFromRestricted,
-    release,
-  };
+export const getIndexer = async () => {
+	const rollcall = await getRollcall();
+	const client = await getClient();
```

Comment:
awaited items benefit from having a try/catch wrapper, in case things go sideways